### DISCUSSION TITLE
Pagination and perPage()

### DISCUSSION THREAD

**Discussion by GlenWarren** [orchidsoftware/platform#2136] [Q&A]
I have not been able to get the `perPage()` function to work.
```
    /**
     * Get the number of models to return per page
     *
     * @return int
     */
    public static function perPage(): int
    {
        return 10;
    }
```
Can anyone give some more explanation on where this block of code should live, and if anything else is required?
I have tried adding it in a few places (eg. screens, tables).

Thanks

**Comment by tabuna** [comment]
Hy @GlenWarren. It resource method (When using the [CRUD package](https://orchid.software/en/docs/packages/crud)) returns the number of records [per page by default](https://orchid.software/en/docs/packages/crud/#pagination). Accordingly, it is indicated in the resource files.
